use archive instead of git to have translations

petitlapin opened the pull request:
Should fix https://github.com/flathub/org.kde.gcompris/issues/3
This is not tested, I adapted the code from another repository.
I can try to set up an environment to test if needed but I'm no expert so not sure how much time it could take me.

Thank you,

Johnny

aleixpol commented:
It should be really simple for you to test. You can see the documentation here: http://docs.flatpak.org/en/latest/first-build.html

petitlapin commented:
Yes, let me test, I though the setup would be more difficult.
Thank you for the link

petitlapin commented:
To explain the change:
* using source tarball instead of git. As we don't use git anymore, we can't use "git submodule" to retrieve box2d.
Instead of disabling it, I added a step to retrieve the box2d git and put it where needed.
* translations are already shipped in the source tarball and are automatically compiled.

I've built it using "flatpak-builder build-dir org.kde.gcompris.json" and run it with "flatpak-builder --run build-dir org.kde.gcompris.json gcompris-qt"
I've changed to Catalan, and it was well translated to Catalan after restart

aleixpol commented on org.kde.gcompris.json:
```diff
@@ -18,12 +18,20 @@
+                    "sha256": "6fcf6451f5214ccacf6df51b4caa53ab996c2dc597dda0e8b7301d9c384a0026"
+                },
+                {
+                    "type": "git",
```
You should put box2d before gcompris-qt, otherwise it won't be found as a dependency. Or is it a runtime dependency?

petitlapin replied:
We detect box2d as soon as it is in the source folder and we build it during the compilation of gcompris.
If we put it before downloading the sources, we won't be able to put it inside the source folder

aleixpol replied:
Ugh right, my bad.

nedrichards commented:
@petitlapin as you're an active upstream contributor I've sent you an invitation to be able to contribute to this repo more easily in future. As you've seen flatpak is pretty easy! Thanks so much for your help!
